fe/fe-core/src/main/java/com/starrocks/catalog/HiveResource.java

```diff
@@ -28,11 +29,18 @@
  */
 public class HiveResource extends Resource {
 
+    private static final String MAX_COMPUTE_ENABLE = "maxcompute.enable";
     // require only one property currently
     private static final String HIVE_METASTORE_URIS = "hive.metastore.uris";
 
+    @SerializedName(value = "isMaxComputeResource")
+    private boolean isMaxComputeResource = false;
+
+    @SerializedName(value = "resourceOptions")
+    private final Map<String, String> resourceOptions = new HashMap<>();
```

Comment:
Do the parameters in resourceOptions needn't be verified?

Reply:
I'm not sure whether the parameters in resourceOptions need be verified or not. We should verified some definite parameter like `hive.metastore.uris`, but the parameters for spark data source is indefinite.

For example:

If we want to connect to MaxCompute throught Spark, we could use the data source provided by [aliyun-emapreduce-datasources](https://github.com/aliyun/aliyun-emapreduce-datasources) which we must judge `accessKeyId `, we could also use the data source provided by [aliyun-maxcompute-data-collectors](https://github.com/aliyun/aliyun-maxcompute-data-collectors) which we must judge `spark.hadoop.odps.access.id`.

So, for me, I think it's better to judge the resourceOptions by spark data source.